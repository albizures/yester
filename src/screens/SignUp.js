import React, { Component } from 'react'
import {
  StyleSheet,
  Text, View, Dimensions,
} from 'react-native'
import PropTypes from 'prop-types'
import colors from '../utils/colors'
import Container from '../components/Container'
import Translate from '../components/Translate'
import Button, {types} from '../components/Button'
import styles from '../styles/common'

class SignUp extends Component {
  static propTypes = {
    navigation: PropTypes.object.isRequired,
  }

  onPressSubscription = () => {
    this.props.navigation.navigate('SignUp2')
  }

  onPressLogin = () => {
    this.props.navigation.navigate('Login')
  }

  render () {
    return (
      <View style={localStyles.container2}>
        <Container>
          <View style={[localStyles.container]}>

            <Text style={[styles.h1, localStyles.margin]}>
              <Translate keyName='common.upperTitle' />
            </Text>
            <Text style={[styles.title, localStyles.margin]}>
              <Translate keyName='signUp.weKnow' />
              <Text style={{fontWeight: 'bold'}}>
                <Translate keyName='signUp.weWant' />
              </Text>
            </Text>
            <Text style={[styles.title, localStyles.margin]}>
              <Translate keyName='signUp.receive' />
            </Text>

            <Button onPress={this.onPressSubscription}
              title='signUp.start' />

            <Text style={[styles.body1, localStyles.margin]}>
              <Translate keyName='signUp.price' />
            </Text>

            <Button onPress={this.onPressLogin}
              title='signUp.logIn'
              type={types.OUTLINED} />

            <View style={[styles.separator, localStyles.margin]} />

            <Text style={[styles.body2, {textAlign: 'left', marginBottom: height * 0.01}]}>
              <Translate keyName='signUp.cancel' />
            </Text>
            <Text style={[styles.body1, {textAlign: 'left'}]}>
              <Translate keyName='signUp.recurring' />
            </Text>

          </View>
        </Container>
        <View style={localStyles.container3}>
          <Text style={[styles.title, {textAlign: 'center'}]}>
            <Translate keyName='signUp.already' />
            <Text style={[{fontWeight: 'bold', textDecorationLine: 'underline'}]}>
              <Translate keyName='signUp.restore' />
            </Text>
          </Text>
        </View>
      </View>
    )
  }
}

let { height, width } = Dimensions.get('window')
const localStyles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    zIndex: 0,
    height: height * 0.92,
    paddingTop: height * 0.075,
    paddingHorizontal: width * 0.08,
  },
  container2: {
    position: 'absolute',
    backgroundColor: colors.white,
  },
  container3: {
    flex: 1,
    position: 'absolute',
    zIndex: 2,
    marginTop: height * 0.92,
    height: height * 0.08,
    width: width,
    backgroundColor: colors.athensGray,
    justifyContent: 'center',
    alignItems: 'center',
  },
  margin: {
    marginVertical: height * 0.03,
  },
})

export default SignUp
