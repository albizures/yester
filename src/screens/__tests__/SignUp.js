import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import { TouchableHighlight } from 'react-native'

import SignUp from '../SignUp'
const navigation = {
  navigate: jest.fn(),
}

describe('src/screens/SignUp.js', () => {
  it('should render', () => {
    const tree = renderer.create(<SignUp navigation={navigation} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  describe('when button to free trial is pressed', () => {
    it('should navigate to SignUp screen', () => {
      const shallowLogin = shallow(<SignUp navigation={navigation} />)
      const shallowTouchableHighlights = shallowLogin.find(TouchableHighlight)
      const shallowButtonToSubscription = shallowTouchableHighlights.at(0)
      const shallowButtonToLogin = shallowTouchableHighlights.at(1)

      shallowButtonToLogin.prop('onPress')()
      shallowButtonToSubscription.prop('onPress')()

      expect(navigation.navigate).toHaveBeenCalledTimes(2)
      expect(navigation.navigate).toHaveBeenCalledWith('Subscription')
      expect(navigation.navigate).toHaveBeenCalledWith('Login')
    })
  })
})
