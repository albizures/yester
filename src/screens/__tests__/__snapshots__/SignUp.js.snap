// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/screens/SignUp.js should render 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "position": "absolute",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
        "paddingTop": 20,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "flex": 1,
            "flexDirection": "column",
            "height": 1227.28,
            "paddingHorizontal": 60,
            "paddingTop": 100.05,
            "zIndex": 0,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#313131",
              "fontFamily": "Karla-Regular",
              "fontSize": 50,
              "fontWeight": "bold",
              "textAlign": "left",
            },
            Object {
              "marginVertical": 40.019999999999996,
            },
          ]
        }
      >
        <Text>
          YESTER
        </Text>
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#313131",
              "fontFamily": "Karla-Regular",
              "fontSize": 20,
              "textAlign": "left",
            },
            Object {
              "marginVertical": 40.019999999999996,
            },
          ]
        }
      >
        <Text>
          We know you have beautiful and interesting story. 
        </Text>
        <Text
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          <Text>
            We want to help you, writing yours.
          </Text>
        </Text>
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#313131",
              "fontFamily": "Karla-Regular",
              "fontSize": 20,
              "textAlign": "left",
            },
            Object {
              "marginVertical": 40.019999999999996,
            },
          ]
        }
      >
        <Text>
          Receive one conversation starter a day. Try it for FREE a month (then just $4.99/month).
        </Text>
      </Text>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            undefined,
            Object {
              "borderRadius": 24,
              "borderWidth": 2,
              "height": 50,
              "justifyContent": "center",
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 3,
                "width": 0,
              },
            },
            Object {
              "backgroundColor": "#4241B3",
              "borderColor": "#4241B3",
            },
            Object {
              "width": 300,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "textAlign": "center",
                },
                Object {
                  "color": "#fff",
                },
              ]
            }
          >
            Start free trial
          </Text>
        </View>
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#313131",
              "fontFamily": "Karla-Regular",
              "fontSize": 16,
              "textAlign": "center",
            },
            Object {
              "marginVertical": 40.019999999999996,
            },
          ]
        }
      >
        <Text>
          $4.99/month after trial ends.
        </Text>
      </Text>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            undefined,
            Object {
              "borderRadius": 24,
              "borderWidth": 2,
              "height": 50,
              "justifyContent": "center",
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 3,
                "width": 0,
              },
            },
            Object {
              "backgroundColor": "#fff",
              "borderColor": "#4241B3",
            },
            Object {
              "width": 300,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "textAlign": "center",
                },
                Object {
                  "color": "#4241B3",
                },
              ]
            }
          >
            Log In
          </Text>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "borderBottomColor": "#4241B3",
              "borderBottomWidth": 1,
            },
            Object {
              "marginVertical": 40.019999999999996,
            },
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#313131",
              "fontFamily": "Karla-Regular",
              "fontSize": 16,
              "fontWeight": "bold",
              "textAlign": "center",
            },
            Object {
              "marginBottom": 13.34,
              "textAlign": "left",
            },
          ]
        }
      >
        <Text>
          Cancel Anytime
        </Text>
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#313131",
              "fontFamily": "Karla-Regular",
              "fontSize": 16,
              "textAlign": "center",
            },
            Object {
              "textAlign": "left",
            },
          ]
        }
      >
        <Text>
          At Yester is recurring subscription. If you decide to purchase a subscription, an initial payment will be...
        </Text>
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#F6F7F9",
        "flex": 1,
        "height": 106.72,
        "justifyContent": "center",
        "marginTop": 1227.28,
        "position": "absolute",
        "width": 750,
        "zIndex": 2,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#313131",
            "fontFamily": "Karla-Regular",
            "fontSize": 20,
            "textAlign": "left",
          },
          Object {
            "textAlign": "center",
          },
        ]
      }
    >
      <Text>
        Already purchased? 
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontWeight": "bold",
              "textDecorationLine": "underline",
            },
          ]
        }
      >
        <Text>
          Restore
        </Text>
      </Text>
    </Text>
  </View>
</View>
`;
